mock/py/mockbuild/buildroot.py

```diff
@@ -497,15 +498,39 @@ def _setup_devices(self):
             kver = os.uname()[2]
             self.root_log.debug("kernel version == %s", kver)
             for i in devFiles:
+                src_path = "/" + i[2]
+                chroot_path = self.make_chroot_path(i[2])
+
+                if util.cmpKernelVer(kver, '2.6.18') >= 0 and src_path == '/dev/ptmx':
+                    continue
+
                 # create node, but only if it exist on host too
                 # except for loop devices, which only show up on the host after they are first used
-                if os.path.exists("/" + i[2]) or "loop" in i[2]:
-                    os.mknod(self.make_chroot_path(i[2]), i[0], i[1])
+                if os.path.exists(src_path) or "loop" in src_path:
+                    try:
+                        os.mknod(chroot_path, i[0], i[1])
```

Comment:
Don't you want to fall-back only if we are in the `USE_NSPAWN` scenario?  I.e. don't fallback for normal `chroot()`?

Reply:
Taking back ... toolbox actually implies normal chroot() inside podman container.  So I should reverse my question: Do we want to fallback if `USE_NSPAWN` is on?  Because in that case mock is run in non-toolbox environment (normal host) - and it might be better to fail on mknod (if that really was failing) instead of trying useless bind mount.

Reply:
See my reply to @xsuchy above.

Reply:
TBH, I view this PR approach (running toolbox/podman wrapper around mock, instead of starting containers by mock) as much more straight-forward way to make mock maintainable in future.  And **really** superior.  While USE_NSPAWN is better than nothing, the current nspawn support isn't complete (--installroot is still not run in container) and it IMO doesn't make much sense to continue with finishing it..  having it dropped one day sounds like good idea to me.

But still, ATM, USE_NSPAWN **is** used by default (and e.g. used by copr build system), and in that case - falling down from mknod to bind-mount doesn't make much sense..   Also, from the other side -- trying mknod when we know in advance it will fail (toolbox case) isn't easy reading.  All that is mixed up with bare host + old-chroot use-case.  Just saying.

But yeah, we seem to know now what is the change about and why - at least now, so +1 from me.